pkg/daemon/certificate_writer.go

```diff
@@ -109,6 +111,24 @@ func (dn *Daemon) syncControllerConfigHandler(key string) error {
 			return err
 		}
 
+		for _, CA := range controllerConfig.Spec.ImageRegistryBundleData {
+			if err := os.MkdirAll(filepath.Join("/etc/docker/certs.d", CA.File), defaultDirectoryPermissions); err != nil {
```

Comment:
nit: some of these paths (and other namespaces, variables below) may be better as consts